great_expectations/rule_based_profiler/helpers/simple_semantic_type_filter.py

```diff
@@ -127,9 +127,23 @@ def _infer_semantic_domain_type_from_table_column_type(
         column_type: str = str(column_types_dict_list[0]["type"]).upper()
 
         semantic_column_type: SemanticDomainTypes
-        if column_type in (
-            {type_name.upper() for type_name in ProfilerTypeMapping.INT_TYPE_NAMES}
-            | {type_name.upper() for type_name in ProfilerTypeMapping.FLOAT_TYPE_NAMES}
+        if (
+            column_type
+            in (
+                {type_name.upper() for type_name in ProfilerTypeMapping.INT_TYPE_NAMES}
```

Comment:
@billdirks Good points/question.  However, I would like to keep all those types defined precisely (we painstakingly listed them) -- as well as the logic -- let it exit on first match (also we might change the logic later).  Thanks.

Reply:
`any` should short circuit and exit early.
If you want separate conditionals could you get rid of the first one (the equality check) since that works is repeated in the next two.

Reply:
@billdirks I just left the `any` (if we change implementation, we can figure out the cleanest code then).  Regarding listing all the types -- we want to keep them listed as our catalog of the types we know and recognize as they are spelled and with casing as they appear in the various backends.  Having said that, this whole inference of semantic type from "physical" type is not very good (this is why the filter is called "simple semantic filter" and users can supply their own).